### ISSUE TITLE
Build warning in share: grep: warning: stray \ before #

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2106]
Imported from SourceForge on 2024-07-05 18:40:16
Created by **[billingd](https://sourceforge.net/u/billingd/)** on 2023-01-30 11:33:45
Original: https://sourceforge.net/p/maxima/bugs/4094

---

There are several harmless build warnings in the share subdirectory.

~~~
grep: warning: stray \ before #
~~~

It is generated in the system call to generate make variable sharefiles.  A web search didn't find any need to escape '#' in an extended grep regex.

The fix is trivial.  Tested  with GNU egrep.  I wonder if it will be a problem with other versions of egrep.

~~~
diff --git a/share/Makefile.am b/share/Makefile.am
index f56124228..5681b5872 100644
--- a/share/Makefile.am
+++ b/share/Makefile.am
@@ -7,8 +7,8 @@ egrep -v 'CVS|Makefile(\.in|\.am)?|^\./Makefile|\.gitignore|\.gitattributes' | \
 egrep -v '/fortran/' | \
 egrep -v 'colnew/ex./' | \
 egrep -v 'lbfgs/.*\.f' | \
-egrep -v '/\.\#' | \
-egrep -v '~|\#' | \
+egrep -v '/\.#' | \
+egrep -v '~|#' | \
 sed -e 's%\./%%' | \
 sort -f -d | \
 sed '$$s/\\$$//' )
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:40:18
Created by **[billingd](https://sourceforge.net/u/billingd/)** on 2023-02-02 13:39:19
Original: https://sourceforge.net/p/maxima/bugs/4094/#4eba

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:40:21
Created by **[billingd](https://sourceforge.net/u/billingd/)** on 2023-02-02 13:39:19
Original: https://sourceforge.net/p/maxima/bugs/4094/#b053

---

Fixed by commit [66e507]

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:40:25
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2023-02-04 21:53:56
Original: https://sourceforge.net/p/maxima/bugs/4094/#688d

---

Well, I reconfigured /bin/sh to point to /bin/bash (via sudo dpkg-reconfigure dash) and I get the same error. For the record the error message is: 

```
Makefile:338: *** unterminated call to function `shell': missing `)'.
```

At this point I'm thinking the error is coming from make, which is trying to piece together the stuff to execute in a shell. 

I updated make to version 4.4 and now it works. Previous version 3.81 was released in 2006, so it's okay to just leave it behind.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:40:28
Created by **[billingd](https://sourceforge.net/u/billingd/)** on 2023-02-05 01:24:45
Original: https://sourceforge.net/p/maxima/bugs/4094/#d588

---

I am happy to revert this patch if it is a problem.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 18:40:32
Created by **[billingd](https://sourceforge.net/u/billingd/)** on 2023-02-07 10:12:51
Original: https://sourceforge.net/p/maxima/bugs/4094/#d2bb

---

Wolfgang Dautermann reverted this in commit [e00f35] with message

    Build system: Escape '#' in $(shell ...) command.
    
    Otherwise the build with GNU Make <= 4.2 fails, see
    https://stackoverflow.com/questions/53836858/
    
    And GNU Make 4.2.1 is included in Ubuntu 20.04 so it may be in
    use for some years, the distribution is supported until 2025.
